tests/adapters_t/batchsystems_t/test_slurm.py

```diff
@@ -0,0 +1,242 @@
+        )
+
+        self.mock_async_run_command.reset_mock()
+
+        run_async(self.slurm_adapter.drain_machine, drone_uuid="VM-1")
+        self.mock_async_run_command.assert_called_with(
+            "scontrol update NodeName=host-10-18-1-1 State=DRAIN Reason='COBalD/TARDIS'"
+        )
+        self.assertIsNone(
+            run_async(self.slurm_adapter.drain_machine, drone_uuid="not_exists")
+        )
+        self.mock_async_run_command.side_effect = CommandExecutionFailure(
+            message="Does not exists", exit_code=1, stderr="Does not exists"
+        )
+        self.assertIsNone(
+            run_async(self.slurm_adapter.drain_machine, drone_uuid="test")
+        )
+
+        # Review: Not working, and I don't know why. What is this testing?
+        #  self.mock_async_run_command.side_effect = CommandExecutionFailure(
+        #      message="Unhandled error", exit_code=2, stderr="Unhandled error"
+        #  )
+        #  with self.assertRaises(CommandExecutionFailure):
+        #      self.assertIsNone(
+        #          run_async(self.slurm_adapter.drain_machine, drone_uuid="test")
+        #      )
+
+        self.mock_async_run_command.side_effect = None
+
+    def test_drain_machine_without_options(self):
+        self.setup_config_mock()
+        self.slurm_adapter = SlurmAdapter()
+
+        run_async(self.slurm_adapter.drain_machine, drone_uuid="VM-1")
+        self.mock_async_run_command.assert_called_with(
+            "scontrol update NodeName=host-10-18-1-1 State=DRAIN Reason='COBalD/TARDIS'"
+        )
+
+    def test_integrate_machine(self):
+        self.assertIsNone(
+            run_async(self.slurm_adapter.integrate_machine, drone_uuid="VM-1")
+        )
+
+    def test_get_resource_ratios(self):
+        self.assertCountEqual(
+            list(run_async(self.slurm_adapter.get_resource_ratios, drone_uuid="VM-1")),
+            [self.cpu_ratio, self.memory_ratio],
+        )
+        self.mock_async_run_command.assert_called_with(self.command)
+
+        self.assertEqual(
+            run_async(self.slurm_adapter.get_resource_ratios, drone_uuid="not_exists"),
+            {},
+        )
+
+    def test_get_resource_ratios_without_options(self):
+        self.setup_config_mock()
+        del self.config.BatchSystem.options
+        self.slurm_adapter = SlurmAdapter()
+
+        self.assertCountEqual(
+            list(run_async(self.slurm_adapter.get_resource_ratios, drone_uuid="VM-1")),
+            [self.cpu_ratio, self.memory_ratio],
+        )
+
+        self.mock_async_run_command.assert_called_with(self.command_wo_options)
+
+    def test_get_allocation(self):
+        self.assertEqual(
+            run_async(self.slurm_adapter.get_allocation, drone_uuid="VM-1"),
+            max([self.cpu_ratio, self.memory_ratio]),
+        )
+        self.mock_async_run_command.assert_called_with(self.command)
+
+    def test_get_machine_status(self):
```

Comment:
I would prefer to use a loop here. Like in https://github.com/MatterMiners/tardis/blob/cefc49c18bcc4d16910f1858793bc59a8dd17501/tests/adapters_t/sites_t/test_slurm.py#L245-L286